backend/app/domains/courses/crud.py

```
@@ -35,7 +46,29 @@ def get_course(db: Session, course_id: int) -> Course | None:
 def get_students_by_course(db: Session, course_id: int) -> List[Student]:
     stmt = (
         select(Student)
-        .join(Register, Register.student_id == Student.id)
+        .join(Register, Register.user_id == Student.id)
         .where(Register.course_id == course_id)
     )
     return db.exec(stmt).all()
+
+
+def delete_course(db: Session, course: Course, current_user: User):
+    deleted_registrations = delete_registrations_by_course_id(db, course.id, current_user)
```

Ruff (0.11.9)

Local variable `deleted_registrations` is assigned to but never used

Remove assignment to unused variable `deleted_registrations`

(F841)